packages/core/src/model.types.ts

```diff
@@ -32,7 +32,7 @@ export interface Model<
   createMachine: (
     config: MachineConfig<TContext, any, TEvent>,
     implementations?: Partial<MachineOptions<TContext, TEvent>>
-  ) => StateMachine<TContext, any, TEvent, any>;
+  ) => StateMachine<TContext, any, TEvent>;
```

Comment:
This resulted in the default type not being chosen here:
https://github.com/statelyai/xstate/blob/141c91cffd1d7c1ec2e82186834cb977b72fb4d4/packages/core/src/State.ts#L93
which in turn resulted in `any` being selected here:
https://github.com/statelyai/xstate/blob/68692c1d11df7a513e68d4ce888603c1ffe41552/packages/core/src/State.ts#L298

The alternative fix here would be to use the default type explicitly here. It would be more explicit but it's probably not worth it at the moment.

Reply:
Thanks for digging into this and fixing it.

Reply:
No problem. That's my job 😍